tests/hooks/test_dbt_hook.py

```diff
@@ -1,54 +1,211 @@
-from unittest import TestCase
-from unittest import mock
-import subprocess
-from airflow_dbt.hooks.dbt_hook import DbtCliHook
+from typing import Union
+from unittest import TestCase, mock
+from unittest.mock import patch
 
+import pytest
+from airflow import AirflowException
+from airflow.hooks.subprocess import SubprocessHook, SubprocessResult
+from pytest import mark
 
-class TestDbtHook(TestCase):
+from airflow_dbt.dbt_command_config import DbtCommandConfig
+from airflow_dbt.hooks.base import generate_dbt_cli_command
+from airflow_dbt.hooks.cli import DbtCliHook
+from airflow_dbt.hooks.google import DbtCloudBuildHook
 
-    @mock.patch('subprocess.Popen')
-    def test_sub_commands(self, mock_subproc_popen):
-        mock_subproc_popen.return_value \
-            .communicate.return_value = ('output', 'error')
-        mock_subproc_popen.return_value.returncode = 0
-        mock_subproc_popen.return_value \
-            .stdout.readline.side_effect = [b"placeholder"]
 
-        hook = DbtCliHook()
-        hook.run_cli('docs', 'generate')
-
-        mock_subproc_popen.assert_called_once_with(
-            [
-                'dbt',
-                'docs',
-                'generate'
-                ],
-            close_fds=True,
-            cwd='.',
-            stdout=subprocess.PIPE,
-            stderr=subprocess.STDOUT
-            )
+@mark.parametrize(
+    ["dbt_bin", "command", "params", "expected_command"], [
+        ("dbt", "run", {}, ["dbt", "run"]),
```

Comment:
I like that you have added new lines to group the test cases on line 58 and below, it makes it easier to follow. Perhaps you could do the same here?